### PR TITLE
IdentitySCIMManagerTest#001

### PR DESCRIPTION
**Pull request by ShanChathusanda93**
Created two test methods in IdentitySCIMManagerTest

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.wso2.carbon.identity.scim2.common.utils.SCIMConfigProcessor;
+
+import java.nio.file.Paths;
+
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertNotNull;
+
+@PrepareForTest({ SCIMCommonUtils.class }) @PowerMockIgnore({ "javax.xml.*",
+        "org.apache.axis2.*" }) public class IdentitySCIMManagerTest extends PowerMockTestCase {
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java]: Fix formatting.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Test public void testGetInstance() throws Exception {
+        assertNotNull(identitySCIMManager, "Returning a null");
+        assertNotNull(identitySCIMManager, "Returning a null");
+    }
+
+    @Test public void testGetEncoder() throws Exception {
+        assertNotNull(identitySCIMManager.getEncoder());
+    }
+}
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java]: Add a newline.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        UserManager userManager = identitySCIMManager.getUserManager();
+        assertNotNull(userManager);
+    }
+
+    @Test
+    public void testGetUserManagerWithException() throws Exception {
+        try {
+            when(SCIMCommonComponentHolder.getRealmService()).thenReturn(null);
+            identitySCIMManager.getUserManager();
+            fail("Expexted a CharonException, but did not throw");
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java]: There is a typo in `Expexted`. Anyway better to use a message as below. 
`getUserManager() method should have thrown a CharonException.`

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        }
+    }
+
+    @Test
+    public void testGetUserManagerWithException2() throws Exception {
+        try {
+            when(SCIMCommonComponentHolder.getRealmService()).thenReturn(realmService);
+            when(mockedTenantManager.getTenantId(anyString())).thenThrow(new UserStoreException());
+            identitySCIMManager.getUserManager();
+            fail("Expexted a CharonException, but did not throw");
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java]: There is a typo in `Expexted`. Anyway better to use a message as below. 
`getUserManager() method should have thrown a CharonException.`

**Comment by mefarazath** [comment]
@ShanChathusanda93 there seems to be a merge conflict.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.fail;
+
+@PrepareForTest({SCIMCommonUtils.class, PrivilegedCarbonContext.class, SCIMCommonComponentHolder.class,CharonConfiguration.class})
+
+public class IdentitySCIMManagerTest extends PowerMockTestCase {
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java:54]: Shall we fix the formatting errors?

**Pull request merged**